### PR TITLE
feat: Exa.TrkX with torchscript backend

### PR DESCRIPTION
**Pull request by benjaminhuth**
This PR is a big update to the Exa.TrkX plugin:

* It now provides both a TorchScript-based and a ONNX-based backend (the ONNX one doesn't really work at the moment due to missing operators in Onnx, however, we should'nt throw away the code for this I think)
* Partial integration of Acts logging instead of `std::cout` based things
* The `TrajectoriesToProtoTracks` algorithm to be able to compare results of the CKF and the Exa.TrkkX track finding
* Update python to support all of this
* The dependency to FRNN is now managed by CMake's `FetchContent` mechanism
* The CI routines are restrucutured
  * no build in the github workflows anymore
  * Build and check both backends in the CI bridge

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,6 +34,8 @@ option(ACTS_BUILD_PLUGIN_ACTSVG "Build SVG display plugin" OFF)
 option(ACTS_BUILD_PLUGIN_CUDA "Build CUDA plugin" OFF)
 option(ACTS_BUILD_PLUGIN_DD4HEP "Build DD4hep plugin" OFF)
 option(ACTS_BUILD_PLUGIN_EXATRKX "Build the Exa.TrkX plugin" OFF)
+option(ACTS_EXATRKX_ENABLE_ONNX "Build the Onnx backend for the exatrkx plugin" OFF)
+option(ACTS_EXATRKX_ENABLE_TORCH "Build the torchscript backend for the exatrkx plugin" ON)
```
[the remark this answers is not in the document]
Comment (benjaminhuth) [reply CMakeLists.txt:38]: What I thought: if we set `ACTS_BUILD_PLUGIN_EXATRKX` to `ON`, there should already one backend built automatically by default, and not requiring to set a second option to be set.

But of course this also could be set to `OFF`, that would let the cmake configuration fail with the comment that at least one backend should be enabled...

File: Plugins/ExaTrkX/include/Acts/Plugins/ExaTrkX/ExaTrkXTiming.hpp [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    ACTS_VERBOSE("4) gnn: " << gnn);
+    ACTS_VERBOSE("5) labeling: " << labeling);
+    ACTS_VERBOSE("6) total: " << total);
+  }
+};
+
+/// @struct ExaTrkXTimeList
+///
+/// @brief Represents a series of Exa.TrkX timing information
+struct ExaTrkXTimeList {
```
[the remark this answers is not in the document]
Comment (benjaminhuth) [reply Plugins/ExaTrkX/include/Acts/Plugins/ExaTrkX/ExaTrkXTiming.hpp:45]: Hmm you're right, I'll remove it for now

File: Examples/Run/Reconstruction/CMakeLists.txt [hunk 3]
```diff
@@ -132,4 +132,4 @@ install(
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 add_subdirectory_if(DD4hep ACTS_BUILD_EXAMPLES_DD4HEP)
-add_subdirectory_if(ExaTrkX ACTS_BUILD_EXAMPLES_EXATRKX)
+#add_subdirectory_if(ExaTrkX ACTS_BUILD_EXAMPLES_EXATRKX)
```
[the remark this answers is not in the document]
Comment (Corentin-Allaire) [reply Examples/Run/Reconstruction/CMakeLists.txt:135]: It was more a comment in the style of : "if you want to remove it just remove do not comment it out". I fully agree with you you shouldn't bother with the binary

File: Plugins/ExaTrkX/src/ExaTrkXTrackFindingOnnx.cpp [hunk 4]
```diff
@@ -233,9 +100,10 @@ void Acts::ExaTrkXTrackFinding::buildEdges(std::vector<float>& embedFeatures,
             std::back_inserter(edgeList));
 }
 
-void Acts::ExaTrkXTrackFinding::getTracks(
-    std::vector<float>& inputValues, std::vector<uint32_t>& spacepointIDs,
-    std::vector<std::vector<uint32_t> >& trackCandidates) const {
+Acts::ExaTrkXTime Acts::ExaTrkXTrackFindingOnnx::getTracks(
+    std::vector<float>& inputValues, std::vector<int>& spacepointIDs,
+    std::vector<std::vector<int> >& trackCandidates, LoggerWrapper) const {
+  ExaTrkXTime timeInfo;
```
[the remark this answers is not in the document]
Comment (benjaminhuth) [reply Plugins/ExaTrkX/src/ExaTrkXTrackFindingOnnx.cpp:107]: I made timing optional, can you have a look again?

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (benjaminhuth) [reply CMakeLists.txt:38]: I will keep it as it is now...

**Review by Corentin-Allaire — APPROVED** [review]
Perfect this looks good to me !

**Comment by acts-project-service** [comment]
The backport to `develop/v19.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-develop/v19.x develop/v19.x
# Navigate to the new working tree
cd .worktrees/backport-develop/v19.x
# Create a new branch
git switch --create backport-1473-to-develop/v19.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick -x --mainline 1 b409627a280fae20ea17663fddb3d74a38d0716b
# Push it to GitHub
git push --set-upstream origin backport-1473-to-develop/v19.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-develop/v19.x
```
Then, create a pull request where the `base` branch is `develop/v19.x` and the `compare`/`head` branch is `backport-1473-to-develop/v19.x`.

**Pull request merged**